gui/src/ruler.cc

```diff
@@ -0,0 +1,362 @@
+#include <cmath>
+#include <iostream>
+#include "ruler.hh"
+
+////////////////////////////////////////////////////////////////////////
+// Ruler
```

Comment:
There's many repeated `if`s in this file both `if(orientation)` and `if(lowerToUpper)`. You might want to look into the "Strategy" pattern, to see if you can eliminate those.